internal/cachecontroller/cache_controller.go

```diff
@@ -190,8 +197,9 @@ func (c *InMemoryCacheController) findChangesAndInvalidate(ctx context.Context,
 }
 
 func (c *InMemoryCacheController) invalidateIteratorCache(storeID string) {
-	// These entries do not need to expire
-	c.cache.Set(storage.GetInvalidIteratorCacheKey(storeID), &storage.InvalidEntityCacheEntry{LastModified: time.Now()}, math.MaxInt)
+	// These entries do not need to expire. We set it as OneYear instead of math.MaxInt so that it could not be interpreted
+	// as negative number.
```

Comment:
```suggestion
	// These entries do not need to expire. 
```

- The godoc I removed will not make sense a year from now so it's best to remove it.

- Instead of making this change here, you could make the change within our caching library wrapper, so that if you give it `math.maxInt` it transforms it to a value that Theine will not overflow on. The benefit will be that any user of the cache will not have to remember about this "gotcha". 

- Please add a unit test for this, here or in our caching wrapper if you take up on my suggestion :)